src/ui/public/query_bar/components/query_bar.tsx

```diff
@@ -533,6 +604,7 @@ export class QueryBarUI extends Component<Props, State> {
             </div>
           </EuiOutsideClickDetector>
         </EuiFlexItem>
+        {this.renderDatePicker()}
         <EuiFlexItem grow={false}>
           <EuiButton
```

Comment:
Why not use the EuiSuperDatePicker's update/refresh button and already includes all the states you need including icons and tooltip? Otherwise, I'd ask that we port over that button to here.

Reply:
Because then there would be 2 update buttons - one from EuiSuperDatePicker and one from QueryBar. Linking the DatePicker changes to the QueryBar update button allows users to set the query and time range and apply the changes in a single action.

Reply:
Right, but I thought the EuiSuperDatePicker's update button was created in a way that you could tap into it's state and onClick functions so that it could be used **instead** of the one that's already there.

Reply:
No. The EuiSuperDatePicker's update button just works on the start/end props, not outside state like query.

Reply:
So its the other way around, if you want update logic to include other stuff then you set `showUpdateButton ` to false and add your own button/logic in the wrapping component

Reply:
Ok, then we'll want to convert the EuiSuperDatePicker's `renderUpdateButton` function to an actual EUI component so that it can replace the query bar's update button. That can be an 'eventually' but the states and tooltips that are there for the date picker should be there when the date picker is in conjunction with the query bar.